go/arrow/array/concat.go

```diff
@@ -42,7 +43,12 @@ func Concatenate(arrs []arrow.Array, mem memory.Allocator) (result arrow.Array,
 
 	defer func() {
 		if pErr := recover(); pErr != nil {
-			err = fmt.Errorf("arrow/concat: unknown error: %v", pErr)
+			switch e := pErr.(type) {
+			case error:
```

Comment:
I don't really know go but why is it important to distinguish between these two error types?  Adding `unknown error` doesn't seem to contribute too much to a user's understanding of the error.

Reply:
using `%w` wraps the error that was panic'd so that someone receiving the error can use `errors.Is` or `errors.As` if they desire to react differently based on the type of the error (such as `arrow.ErrInvalid` vs `arrow.ErrIndex` vs `arrow.ErrNotImplemented` etc...) basically it creates an error chain

In the case of the `"unknown error"` it means that we had a panic with a string rather than a proper error instance. Though I agree that the text "unknown error" doesn't contribute much to a user's understanding here, so i'll just get rid of that.